src/display/optional_content_config.js

```diff
@@ -229,12 +230,26 @@ class OptionalContentConfig {
     return true;
   }
 
-  setVisibility(id, visible = true) {
+  setVisibility(id, visible = true, preserveRB = true) {
     const group = this.#groups.get(id);
     if (!group) {
       warn(`Optional content group not found: ${id}`);
       return;
     }
+
+    // if my visibility is about to be set to `true' and if I belong to one or
+    // more radiobutton groups, hide all other OCGs in these radiobutton groups,
+    // provided that radio-button state relationships are to be preserved
+    if (visible && group.myRbGroups.length && preserveRB) {
```

Comment:
Let's re-order this slightly, such that we do the "simpler" checks first.
```suggestion
    if (preserveRB && visible && group.myRbGroups.length) {
```